### PR TITLE
Change homepage sections for activation

### PR DESCRIPTION
**Pull request by ranquild**
Epic: https://github.com/metabase/metabase/issues/19119

Changes:
- Renames `Overworld` to `Homepage`
- Splits `Homepage` into individual sections
- Adds `Add database` link to `Our data` section
- Adds `Add database` banner to `Start here` section
- Some small visual improvements
- Adds unit and cypress tests

Screenshots:
Screen Shot 2021-12-02 at 5 21 06 pm
Screen Shot 2021-12-02 at 5 21 12 pm

Happy path:
- Run a fresh Metabase instance
- Complete the setup flow
- Check that the new database banner and `Add database` link are on the homepage

Try:
- Removing and restoring the sample database
- Adding a non-sample database
- Hiding homepage sections
- Pinning dashboards
- Creating collections

### CODE REVIEW

**Comment by mazameli** [comment]
A hover state here would be a good touch:

image

**Comment by mazameli** [comment]
It would be cool it the `invite a teammate` link here just opened up the invite-user modal in place without having to take me to the Admin section, but that's likely future polish.

image

File: frontend/src/metabase/home/homepage/components/StartSection/StartSection.jsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        <BannerModelIcon name="database" />
+      </BannerIconContainer>
+      <BannerContent>
+        <BannerTitle>{t`Connect your data to get the most out of Metabase`}</BannerTitle>
+        <BannerDescription>
+          {jt`If you need help, you can ${(
+            <ExternalLink href={userUrl}>{t`invite a teammate`}</ExternalLink>
+          )} or ${(
+            <ExternalLink href={docsUrl}>
+              {t`check out our set up guides`}
```
Comment (mazameli) [inline frontend/src/metabase/home/homepage/components/StartSection/StartSection.jsx:105]:
```suggestion
              {t`check out our setup guides`}
```
```suggestion
              {t`check out our set up guides`}
```

File: frontend/src/metabase/home/homepage/components/StartSection/StartSection.jsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    </CardRoot>
+  );
+};
+
+DashboardCard.propTypes = cardProps;
+
+const DatabaseBanner = () => {
+  const userUrl = Urls.newUser();
+  const databaseUrl = Urls.newDatabase();
+  const docsUrl = Settings.docsUrl("getting-started");
```
Comment (mazameli) [inline frontend/src/metabase/home/homepage/components/StartSection/StartSection.jsx:91]: I think this should go to https://www.metabase.com/docs/latest/setting-up-metabase.html or https://www.metabase.com/docs/latest/administration-guide/01-managing-databases.html

**Comment by agnes-pinhanelli** [comment]
Database button hover state [added to Figma](https://www.figma.com/file/E3zVOq6PgSeRyqsOmfdPE7/Remind-users-to-connect-their-data-after-set-up?node-id=326%3A2802)

**Comment by agnes-pinhanelli** [comment]
@mazameli Yes, I actually worked on a version with the 'invite a teammate' modal for the homepage but we decided to save it for a future version indeed. But we have the drafts anyway 🚀  

[Screen Shot 2021-12-03 at 10 23 36](https://user-images.githubusercontent.com/95483664/144609659-bb20fea9-c2aa-48c7-a614-1875e5974499.png)

File: frontend/src/metabase/home/homepage/components/StartSection/StartSection.jsx (hunk 2 above)
Comment (cdeweyx) [reply frontend/src/metabase/home/homepage/components/StartSection/StartSection.jsx]: Good catch, let's do https://www.metabase.com/docs/latest/administration-guide/01-managing-databases.html

**Comment by ranquild** [comment]
@mazameli Fixed the link and the hover state, thanks!

**Review by mazameli — APPROVED** [review]

**Review by cdeweyx — APPROVED** [review]

File: frontend/src/metabase/home/homepage/components/Homepage/Homepage.jsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  showXrays,
+  showPinMessage,
+  onHideData,
+  onHideXrays,
+  onHidePinMessage,
+}) => {
+  return (
+    <LandingRoot>
+      <GreetingSection user={user} />
+      {databases && collections && dashboards && (
```
Comment (alxnddr) [inline frontend/src/metabase/home/homepage/components/Homepage/Homepage.jsx:40]: So now everything waits to be shown until data for all sections is loaded. Is there an opportunity to split that?

**Review by alxnddr — APPROVED** [review]
Great job, nice and clean!

File: frontend/src/metabase/home/homepage/components/Homepage/Homepage.jsx (hunk 3 above)
Comment (ranquild) [reply frontend/src/metabase/home/homepage/components/Homepage/Homepage.jsx:40]: We need this data to understand what content to show in each section, so unfortunately there is not much we can do here. The only thing that is not required is xrays content, and it's lazy loaded.

**Pull request merged**